### PR TITLE
Implement a Gecko runtime telemetry delegate using Glean

### PR DESCRIPTION
**Pull request by Dexterp37**
The implemented delegate redirects all streaming telemetry calls to the Glean SDK metrics defined in the `metrics.yaml` file extracted from the GeckoView AAR the engine-gecko-* component depends on.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.

### CODE REVIEW

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/RemoveMe.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package mozilla.components.browser.engine.gecko;
```
Comment (Dexterp37) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/RemoveMe.java]: NOTE: This needs to be removed before merging.

**Comment by Dexterp37** [comment]
Also r? @mdboom for the GV Glean SDK context and @chutten for docs.

File: components/browser/engine-gecko-nightly/README.md [hunk 2]
```diff
@@ -12,6 +12,25 @@ Use Gradle to download the library from [maven.mozilla.org](https://maven.mozill
 implementation "org.mozilla.components:browser-engine-gecko-nightly:{latest-version}"
 ```
 
+### Integration with the Glean SDK
+
+The [Glean SDK](../../../components/service/glean/README.md) can be used to collect Gecko telemetry.
```
Comment (chutten) [inline components/browser/engine-gecko-nightly/README.md]:
```suggestion
The [Glean SDK](../../../components/service/glean/README.md) can be used to collect [Gecko Telemetry](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/index.html).
```

File: components/browser/engine-gecko-nightly/README.md [hunk 3]
```diff
@@ -12,6 +12,25 @@ Use Gradle to download the library from [maven.mozilla.org](https://maven.mozill
...
+Applications using both this component and the Glean SDK should setup the Gecko telemetry delegate
```
Comment (chutten) [inline components/browser/engine-gecko-nightly/README.md]:
```suggestion
Applications using both this component and the Glean SDK should setup the Gecko Telemetry delegate
```

**Comment by Dexterp37** [comment]
Ouch, this is blocked again :( On #4079 this time :(

**Pull request merged**